components/wso2is.notification.event.handlers/src/main/java/org/wso2/is/notification/EventSender.java

```diff
@@ -0,0 +1,99 @@
+
+/*
+ *   Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.is.notification;
+
+import com.google.gson.Gson;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.wso2.is.notification.event.Event;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *Utility class to push events.
+ */
+public class EventSender {
+
+    private static final Log log = LogFactory.getLog(EventSender.class);
+    private static final EventSender instance = new EventSender();
+    private static final ThreadPoolExecutor threadPoolExecutor = new ThreadPoolExecutor(200, 500, 100L,
+            TimeUnit.SECONDS,
+            new LinkedBlockingDeque<Runnable>() {
+            });
+
+    private EventSender() {
+
+    }
+
+    public void execute(EventSender.EventRunner eventRunner) {
```

Comment:
Wouldn't the EventSender be simplified if we expose a publishEvent(Event event) method instead of an execute(EventRunner)? With the new suggestion EventRunner is never visible outside the EventSender. And Notification URL is read from EventSender upon initialisation (or may get injected sometime later.). So all what client code has to call is EventSender.getInstance().publisherEvent(Event)